interceptor/interceptor.go

```diff
@@ -31,3 +32,15 @@ type TraceInterceptor interface {
 	StartTraceInterception(ctx context.Context, destination spanreceiver.SpanReceiver) error
 	StopTraceInterception(ctx context.Context) error
 }
+
+// A MetricsInterceptor is an "arbitrary data"-to-"metric proto" converter.
+// Its purpose is to translate data from the wild into metric proto accompanied
+// by a *commonpb.Node to uniquely identify where that data comes from.
+// MetricsInterceptor feeds a metricsreceiver.MetricsReceiver with data.
+//
+// For example it could be Prometheus data source which translates
+// Prometheus metrics into *metricpb.Metric-s.
+type MetricsInterceptor interface {
+	StartMetricInterception(ctx context.Context, destination metricsreceiver.MetricsReceiver) error
```

Comment:
ditto here about Metrics and below too :)

Reply:
Overlooked this one - fixed.